src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -2183,7 +2052,12 @@ private static string GetChangeEmailPurpose(string newEmail)
                     errors.AddRange(result.Errors);
                 }
             }
-            return errors.Count > 0 ? IdentityResult.Failed(errors.ToArray()) : IdentityResult.Success;
+            if (errors.Count > 0)
+            {
+                Logger.LogWarning($"User {await GetUserIdAsync(user)} validation failed: {string.Join(";", errors.Select(e => e.Code))}.");
```

Comment:
Based on previous changes, we do _not_ want to do any formatting up front. This passes an opaque string into the logging system and some of the logging providers can do useful things with the raw data. So, use the logging-specific format strings and pass the values in.

(@Eilon you wanted to be mentioned)